packages/client/src/comp/util/Signature.tsx

```diff
@@ -24,25 +25,42 @@ type Props = React.PropsWithChildren<{
 
 export const Signature: React.FC<Props> = ({ setSignature, setSignatureType, signatureType }) => {
   const { query } = useAppHistory()
-  return <Margin>
-    <Switchable visibleSwitchButton={query['case'] !== 'upload'} signatureType={signatureType} setSignatureType={setSignatureType}>
-    {
-      (signatureType) => {
-        if (signatureType === 'canvas') {
-          return <div>
-            <div>
-              <Canvas setSignature={setSignature} width={width} height={height}/>
+  const { locale } = AddressContainer.useContainer()
+  const stateSignatureType = eligibleSignatureType(locale?.state)
+
+  React.useEffect(() => {
+    // Ensures signatureType is the same as the one from stateSignatureType
+    if (stateSignatureType !== 'both') {
+      setSignatureType(stateSignatureType)
+    }
+  }, [stateSignatureType, setSignatureType])
+
+  if (stateSignatureType === 'both') {
+    return <Margin>
+      <Switchable visibleSwitchButton={query['case'] !== 'upload'} signatureType={signatureType} setSignatureType={setSignatureType}>
```

Comment:
@marcoacfilho: are we using `query['case'] !== 'upload'`?  I don't think we are (I don't see this PR merged yet from the old repo): https://app.asana.com/0/1183067649239362/1187372980312433/f

Reply:
Yeah we are not using it anymore! the new code is `<Switchable visible={signatureType !== 'Upload'}>`